cou/steps/hypervisor.py

```diff
@@ -159,6 +159,13 @@ def azs(self) -> AZs:
                     logger.debug("skipping machine %s", unit.machine.machine_id)
                     continue
 
+                unit_os_release = app.unit_max_os_version(unit)
+                if unit_os_release >= target:
+                    logger.debug(
+                        "skipping unit %s is already on %s", unit.name, str(unit_os_release)
```

Comment:
Is `str(...)` really necessary?
```suggestion
                        "skipping unit %s is already on %s", unit.name, unit_os_release
```